dashboard/app/controllers/api_controller.rb

```diff
@@ -38,6 +37,32 @@ def google_classrooms
     end
   end
 
+  def google_classrooms
+    query_google_classroom_service do |service|
+      response = service.list_courses(teacher_id: 'me')
+      render json: response.to_h
+    end
+  end
+
+  def import_google_classroom
+    course_id = params[:course_id].to_s
+
+    query_google_classroom_service do |service|
+      students = service.list_course_students(course_id).students.map do |student|
+        OmniAuth::AuthHash.new(
+          uid: student.user_id,
+          provider: 'google_oauth2',
+          info: {
+            name: student.profile.name.full_name
+          },
+        )
+      end
+
+      section = Section.from_omniauth(course_id, Section::LOGIN_TYPE_GOOGLE_CLASSROOM, students, current_user.id)
```

Comment:
What's the format of this google-provided course_id? Do we intend this to be used on a URL like the existing join URLs with the 6-char codes, or does it not matter if it's human-readable?

I like the prefix idea. The section codes are [upper-case consonant strings](https://github.com/code-dot-org/code-dot-org/blob/2f7d0a08f836ef1268aeec0964764d75268b42e6/lib/cdo/code_generation.rb#L6) so a prefix with another symbol that can't conflict, e.g. `G-` would be great.